feat: Add resume position tracking for all play types

hoffie opened the pull request:
The tracking is active by default, but the resuming has to be enabled explicitly, either by setting
  `playermpd.play_position_tracking.resume_by_default: true`
or by calling the play_* functions with the new resume=True kwarg.

Related: #1946

Reasons for draft status / checklist before merging:
- [ ] #2330
- [ ] Removal of existing partial resume support
- [ ] Final round of testing

I've been using most of the code on my Phoniebox without problems. The code has seen some adjustments by now though, so it hasn't been tested as-is. Other than that, it's ready for review. I will do a final round of testing when the reviews are done.

Previous review by @pabera (most of the actionable items should have been incorporated by now): https://github.com/hoffie/RPi-Jukebox-RFID/commit/c4805cebb848e8aaafd69c43f497b09436e67169

s-martin commented on resources/default-settings/jukebox.default.yaml:
```diff
@@ -87,6 +87,10 @@ playermpd:
   mpd_conf: ~/.config/mpd/mpd.conf
+  play_position_tracker:
+    resume_by_default: false
+    file: ../../shared/settings/play_positions.json
```
Just a remark: it's not a setting from my point of view, so maybe `shared/logs` is a better place?

hoffie replied:
Changed.

s-martin commented on resources/default-settings/jukebox.default.yaml:
```diff
@@ -87,6 +87,10 @@ playermpd:
     update_on_startup: true
     check_user_rights: true
   mpd_conf: ~/.config/mpd/mpd.conf
+  play_position_tracker:
```
May just `resume` is a more intuitive name for users than the class name?

hoffie replied:
Yeah, I thought the same right when seeing the diff of the PR. I guess I'm going to change the config option and possibly also the class name.

hoffie replied:
Changed.

hoffie committed:
feat: Add resume position tracking

The tracking is active by default, but the resuming has to be
enabled explicitly, either by setting
  playermpd.resume.resume_by_default: true
or by calling the play_* functions with the new resume=True kwarg.

Related: #1946
